bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/PhilipsAirHandler.java

```diff
@@ -0,0 +1,342 @@
+            this.getConfig().put(PhilipsAirConfiguration.CONFIG_DEF_MODEL_ID, getAirPurifierConfig().getModelid());
+            this.getConfig().put(PhilipsAirConfiguration.CONFIG_KEY, getAirPurifierConfig().getKey());
+            ThingHandlerCallback callback = getCallback();
+            if (callback != null) {
+                callback.configurationUpdated(thing);
+            }
+        }
+
+        if (filters != null) {
+            this.filters = filters;
+        }
+
+        return data != null || info != null || filters != null;
+    }
+
+    private void updateChannels() {
+        if (getCallback() != null) {
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                if (ChannelKind.STATE.equals(channel.getKind()) && isLinked(channelUID)) {
+                    updateChannel(channelUID, currentData, deviceInfo, filters);
+                }
+            }
+        }
+    }
+
+    protected void updateChannel(ChannelUID channelUID, @Nullable PhilipsAirPurifierData data,
+            @Nullable PhilipsAirPurifierDevice deviceInfo, @Nullable PhilipsAirPurifierFilters filters) {
+        if (getCallback() != null && isLinked(channelUID)) {
+            Object value;
+            try {
+                value = getValue(channelUID.getAsString(), data, deviceInfo, filters);
+            } catch (Exception e) {
+                logger.debug("AirPurifier doesn't provide {} measurement", channelUID.getAsString().toUpperCase());
+                return;
+            }
+
+            State state = org.eclipse.smarthome.core.types.UnDefType.NULL;
+            if (value instanceof OnOffType) {
+                state = (OnOffType) value;
+            } else if (value instanceof QuantityType<?>) {
+                state = (QuantityType<?>) value;
+            } else if (value instanceof Integer) {
+                state = new DecimalType(BigDecimal.valueOf(((Integer) value).longValue()));
+            } else if (value instanceof String) {
+                state = new StringType(value.toString());
+            } else if (value != null) {
+                logger.warn("Update channel {}: Unsupported value type {}", channelUID,
+                        value.getClass().getSimpleName());
+            }
+
+            updateState(channelUID, state);
+        }
+    }
+
+    @Override
+    protected void updateState(ChannelUID channelUID, State state) {
+        super.updateState(channelUID, state);
+    }
+
+    public @Nullable Object getValue(String channelId, @Nullable PhilipsAirPurifierData data,
+            @Nullable PhilipsAirPurifierDevice deviceInfo, @Nullable PhilipsAirPurifierFilters filters)
+            throws Exception {
+        String[] fields = StringUtils.split(channelId, ":");
+        String field = fields[fields.length - 1];
+        if (field.contains("#")) {
+            field = field.split("#")[1];
+        }

```

Comment:
`ChannelUID` has a method that will return the id without the group element.

```suggestion
    public @Nullable Object getValue(ChannelUID channelUID, @Nullable PhilipsAirPurifierData data,
            @Nullable PhilipsAirPurifierDevice deviceInfo, @Nullable PhilipsAirPurifierFilters filters)
            throws Exception {
        String field = channelUID.getIdWithoutGroup();
```